docs/contributing.md

```diff
@@ -1,16 +1,89 @@
-    * apply format running `make lint`
-        * For examples: `make -C examples lint`
-        * For modules: `make -C modules lint`
-    * verify all tests are passing. Build and test the project with `make test-all` to do this.
-        * For a given module or example, go to the module or example directory and run `make test`.
+`Testcontainers for Go` is open source, and we love to receive contributions from our community — you!
+
+There are many ways to contribute, from writing tutorials or blog posts, improving the documentation, submitting bug reports and feature requests, or writing code for the core library or for a technology module.
+
+In any case, if you like the project, please star the project on [Github](https://github.com/testcontainers/testcontainers-go/stargazers) and help spread the word :)
+Also join our [Slack workspace](http://slack.testcontainers.org) to get help, share your ideas, and chat with the community.
+
+## Questions
+
+GitHub is reserved for bug reports and feature requests; it is not the place for general questions.
+If you have a question or an unconfirmed bug, please visit our [Slack workspace](https://testcontainers.slack.com/);
+feedback and ideas are always welcome.
+
+## Code contributions
+
+If you have a bug fix or new feature that you would like to contribute, please find or open an [issue](https://github.com/testcontainers/testcontainers-go/issues) first.
+It's important to talk about what you would like to do, as there may already be someone working on it,
+or there may be context to be aware of before implementing the change.
+
+Next would be to fork the repository and make your changes in a feature branch. **Please do not commit changes to the `main` branch**,
+otherwise we won't be able to contribute to your changes directly in the PR.
+
+### Submitting your changes
+
+Please just be sure to:
+
+* follow the style, naming and structure conventions of the rest of the project.
+* make commits atomic and easy to merge.
+* use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for the PR title. This will help us to understand the nature of the changes, and to generate the changelog after all the commits in the PR are squashed.
+* use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for your commit messages, as it improves the readability of the commit history, and the review process.
+* unless necessary, please try to **avoid pushing--force** to the branch you submitted a PR from, as it makes it harder to review the changes from a given previous state.
+* verify all tests are passing. Build and test the project with `make test-all` to do this.
+    * For a given module or example, go to the module or example directory and run `make test`.
```

Comment:
suggestion: add a note about the fact this flags a false failure on mac due to a warngin about a malformed LC_DYSYMTAB, which can be ignored.
> === Errors
ld: warning: '/private/var/folders/3y/8hbf585d4yl6f8j5yzqx6wz80000gn/T/go-link-2319589277/000018.o' has malformed LC_DYSYMTAB, expected 98 undefined symbols to start at index 1626, found 95 undefined symbols starting at index 1626